src/Storages/MergeTree/Compaction/MergeSelectorApplier.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <Storages/MergeTree/Compaction/PartProperties.h>
+#include <Storages/MergeTree/Compaction/MergeSelectors/TTLMergeSelector.h>
+#include <Storages/MergeTree/MergeTreeSettings.h>
+#include <Storages/MergeTree/MergeType.h>
+
+#include <Storages/StorageInMemoryMetadata.h>
+
+#include <Common/Logger.h>
+
+#include <optional>
+
+namespace DB
+{
+
+struct MergeSelectorChoice
+{
+    PartsRange range;
+    MergeType merge_type;
+};
+
+class MergeSelectorApplier
```

Comment:
Do we really need class here? Maybe just a function?

Reply:
Function will require all parameters such as aggressive, etc. to be passed through MergerMutator